server/src/test/java/org/opensearch/rest/action/admin/cluster/RestPutDecommissionActionTests.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.admin.cluster;
+
+import org.junit.Before;
+import org.opensearch.action.admin.cluster.decommission.awareness.put.PutDecommissionRequest;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.test.rest.FakeRestRequest;
+import org.opensearch.test.rest.RestActionTestCase;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class RestPutDecommissionActionTests extends RestActionTestCase {
+
+    private RestPutDecommissionAction action;
+
+    @Before
+    public void setupAction() {
+        action = new RestPutDecommissionAction();
+        controller().registerHandler(action);
+    }
+
+    public void testCreateRequest() throws IOException {
+        Map<String, String> params = new HashMap<>();
+        params.put("awareness_attribute_name", "zone");
+        params.put("awareness_attribute_value", "zone-1");
+        params.put("timeout", "10s");
+
+        RestRequest deprecatedRequest = buildRestRequest(params);
+
+        PutDecommissionRequest request = action.createRequest(deprecatedRequest);
+        assertEquals(request.getDecommissionAttribute().attributeName(), "zone");
+        assertEquals(request.getDecommissionAttribute().attributeValue(), "zone-1");
+        assertEquals(request.getTimeout(), TimeValue.timeValueSeconds(10L));
+    }
+
+    public void testCreateRequestWithDefaultTimeout() throws IOException {
+        Map<String, String> params = new HashMap<>();
+        params.put("awareness_attribute_name", "zone");
+        params.put("awareness_attribute_value", "zone-1");
+
+        RestRequest deprecatedRequest = buildRestRequest(params);
+
+        PutDecommissionRequest request = action.createRequest(deprecatedRequest);
+        assertEquals(request.getDecommissionAttribute().attributeName(), "zone");
+        assertEquals(request.getDecommissionAttribute().attributeValue(), "zone-1");
+        assertEquals(request.getTimeout(), TimeValue.timeValueSeconds(300L));
+    }
+
+    private FakeRestRequest buildRestRequest(Map<String, String> params) {
+        return new FakeRestRequest.Builder(xContentRegistry()).withMethod(RestRequest.Method.GET)
```

Comment:
Shouldn't the method be PUT

Reply:
Thanks. fixed it and added unit test for same